src/app/application.cpp

```diff
@@ -495,7 +497,22 @@ int Application::exec(const QStringList &params)
     GuiIconProvider::initInstance();
 #endif
 
-    BitTorrent::Session::initInstance();
+    try {
+        BitTorrent::Session::initInstance();
+    }
+    catch (const RuntimeError &err) {
+#ifdef DISABLE_GUI
+        fprintf(stderr, "%s", qUtf8Printable(err.message()));
+#else
+        QMessageBox msgBox {QMessageBox::Critical};
```

Comment:
1. Sorry for asking for this but can you give me a quick lesson about the curly-braces initialization? I know that when using parentheses it chooses one of the available constructors. About the curly braces, I was under the impression that it chose either the appropriate constructor signature or preferably the `initializer_list<>` constructor. But I can't find either on the online QMessageBox docs. I mean that the Qt docs don't have a constructor signature that takes only a `QMessageBox::Icon` var nor a `initializer_list<>`  constructor.
2. Why not use `QMessageBox::critical()` directly?

Reply:
As I said in the affected Issue it is just a sketch to present a General idea. It has not yet been tested or even compiled. I had no time to finish it.
1. I was wrong, there is no such constructor.
2. It's non recommended way.

Reply:
oh ok. FYI, this idea is what I had in mind too. So am waiting for the final version of the PR.
In the meantime, maybe prefix the PR title with `[WIP]`?